src/lib/dot-matrix/DotMatrix.tsx

```diff
@@ -23,28 +23,50 @@ const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
     legendPosition = DEFAULT_LEGEND_POSITION
   } = props;
 
+  const [width, setWidth] = useState<number>(0);
   const [data, total, overlappingValues] = useDotMatrix(dataPoints, dimensions);
+
+  useEffect(() => {
+    findChartContainerWidth();
+  }, []);
+  useEffect(() => {
```

Comment:
Can we leave some space between the useEffects and functions? Also check if we can customise the formatter for this